doc/sphinx/addendum/implicit-coercions.rst

```diff
@@ -56,51 +56,34 @@ A name ``f`` can be declared as a coercion between a source user-defined class
 conditions holds:
 
  * ``D`` is a user-defined class, then the type of ``f`` must have the form
-   :g:`forall (x₁:A₁)..(xₙ:Aₙ)(y:C x₁..xₙ), D u₁..uₘ` where :math:`m`
+   :g:`forall (x₁:A₁)..(xₖ:Aₖ)(y:C v₁..vₙ), D u₁..uₘ` where :math:`m`
    is the number of parameters of ``D``.
  * ``D`` is ``Funclass``, then the type of ``f`` must have the form
-   :g:`forall (x₁:A₁)..(xₙ:Aₙ)(y:C x₁..xₙ)(x:A), B`.
+   :g:`forall (x₁:A₁)..(xₖ:Aₖ)(y:C v₁..vₙ)(x:A), B`.
  * ``D`` is ``Sortclass``, then the type of ``f`` must have the form
-   :g:`forall (x₁:A₁)..(xₙ:Aₙ)(y:C x₁..xₙ), s` with ``s`` a sort.
+   :g:`forall (x₁:A₁)..(xₖ:Aₖ)(y:C v₁..vₙ), s` with ``s`` a sort.
 
-We then write :g:`f : C >-> D`. The restriction on the type
-of coercions is called *the uniform inheritance condition*.
+We then write :g:`f : C >-> D`. When the :g:`x₁..xₖ` are exactly the
+:g:`v₁..vₙ` (in the same order), the coercion is said to satisfy the
+*uniform inheritance condition*. It is advisable to satisfy it, so
+that the :ref:`test for ambiguous coercion paths <ambiguous-paths>`
+is complete.
 
```

Comment:
```suggestion
We then write :g:`f : C >-> D`. When the :g:`x₁..xₖ` are exactly the
:g:`v₁..vₙ` (in the same order), the coercion is said to satisfy the
*uniform inheritance condition*. When possible, we recommend using coercions
that satisfy this condition in order to NEW WORDING.

```

Could you state the consequences of an ambiguous coercion path here?  My guess is that it means there are multiple possible coercions, in which case Coq may not pick the appropriate one.  The description for "test for ambiguous coercion paths" should be here (main descriptive text) rather than somewhat buried in the warnings you've added.  The warnings can hyperlink to here.

If you think the term is important, maybe change `*uniform inheritance condition*` to ```:gdef:`uniform inheritance condition` ```so it appears in the glossary.  Then you can hyperlink to the term with ```:term:`uniform inheritance condition` ```.

Reply:
Thanks, updated (but feel free to criticize if it's not clear enough).